### ISSUE TITLE
Default Mapper Issue

### ISSUE THREAD

**Issue by ReadingSteiner122** [StanfordLegion/legion#1287]
While writing my Regent code, we're running into issues with the Default Mapper. We've tried compiling with `-ll:csize` with no luck. We're running the code on an Ubuntu 20.04 docker container. Please give some pointers on what to do further.
Here is the error message:

```
[0 - 7fdfc8f0a780]   66.700623 {5}{default_mapper}: Default mapper failed allocation of size 614400 bytes for region requirement of inline mapping in task q_derivatives (UID 27827) in memory 1e00000000000000 (SYSTEM_MEM) for processor 1d00000000000001 (LOC_PROC). This means the working set of your application is too big for the allotted capacity of the given memory under the default mapper's mapping scheme. You have three choices: ask Realm to allocate more memory, write a custom mapper to better manage working sets, or find a bigger machine.
terra: /home/legion/runtime/mappers/default_mapper.cc:2661: virtual void Legion::Mapping::DefaultMapper::map_inline(Legion::Mapping::MapperContext, const Legion::InlineMapping&, const Legion::Mapping::Mapper::MapInlineInput&, Legion::Mapping::Mapper::MapInlineOutput&): Assertion `false' failed.
```

**Comment by rupanshusoi** [comment]
Did you pass an argument to `-ll:csize`? That argument is the number of megabytes you require. E.g. `-ll:csize 4096`.

**Comment by ReadingSteiner122** [comment]
Yup, ran into the same error for `-ll:csize 1024`, `-ll:csize 2048` and `-ll:csize 3000`.

**Comment by elliottslaughter** [comment]
Is there any configuration of this application that runs to completion? E.g., smaller problem size, shorter duration, etc.

If there is, then the next most helpful step would likely be to run with Legion Prof: https://legion.stanford.edu/profiling/#legion-prof

Legion Prof can help diagnose where the memory is going and whether you have e.g., a couple large regions or have many regions being created in a loop. The region your error message mentions specifically is only 600KB, so there either will need to be a large number of such regions or a different, much larger region in order to trigger a message like this.

Feel free to share the Legion Prof results here if you want and we can take a look at it too.

**Comment by ReadingSteiner122** [comment]
We're not running into this issue after updating to the latest commit of the control replication branch.

I'll get back to you if we run into more issues.

Thank you for your help.

**Comment by ReadingSteiner122** [comment]
Unfortunately, after further runs, we've run into this issue again. 
We'll make a Profile report and share the results with you.
